ui/main/src/app/modules/usercard/usercard.component.ts

```diff
@@ -515,14 +514,22 @@ export class UserCardComponent implements OnInit {
 
     private getTimeSpans(specificInformation, startDate, endDate): TimeSpan[] {
         let timeSpans = [];
-        if (!!specificInformation.viewCardInAgenda) {
-            if (!!specificInformation.recurrence) timeSpans = [new TimeSpan(startDate, endDate, specificInformation.recurrence)];
-            else timeSpans = [new TimeSpan(startDate, endDate)];
+        if (!!specificInformation.timeSpans) {
+            specificInformation.timeSpans.forEach( ts => {
+                if (!!specificInformation.recurrence) 
+                    this.opfabLogger.warn("Using deprecated field 'specificInformation.recurrence'. Use 'specificInformation.timeSpan' field instead to configure timespans", LogOption.LOCAL);
+
+                timeSpans.push(new TimeSpan(ts.startDate, ts.endDate, ts.recurrence));
+            });
+        } else if (!!specificInformation.viewCardInAgenda) {
+            if (!!specificInformation.recurrence) {
+                this.opfabLogger.warn("Using deprecated field 'specificInformation.recurrence'. Use 'specificInformation.timeSpan' field instead to configure timespans", LogOption.LOCAL);
+                timeSpans = [new TimeSpan(startDate, endDate, specificInformation.recurrence)];
+            } else timeSpans = [new TimeSpan(startDate, endDate)];
```

Comment:
I suggest having at the beginning the deprecated control  (` if (!!specificInformation.recurrence)  ... ` )  instead of repeating it in each if